client/my-sites/site-settings/start-over.jsx

```diff
@@ -113,7 +118,50 @@ function SiteResetCard( {
 	const dispatch = useDispatch();
 
 	const { data } = useSiteResetContentSummaryQuery( siteId );
+	const {
+		data: status,
+		refetch: refetchResetStatus,
+		isFetching: isFetchingResetStatus,
+	} = useSiteResetStatusQuery( siteId );
+	const [ resetStatus, setResetStatus ] = useState( status?.status ?? 'completed' );
+	const [ wasResetting, setWasReseting ] = useState( false );
 	const [ isDomainConfirmed, setDomainConfirmed ] = useState( false );
+	const [ resetProgress, setResetProgress ] = useState( 0.1 );
+
+	useEffect( () => {
+		let interval = 1;
+		if ( ! siteId || resetStatus === 'completed' ) {
+			return;
+		}
+		if ( ! isFetchingResetStatus ) {
+			interval = setInterval( async () => {
+				const {
+					data: { status: latestStatus },
+				} = await refetchResetStatus();
+				setResetStatus( latestStatus );
+			}, 3000 );
+		}
+		return () => clearInterval( interval );
+	}, [ isFetchingResetStatus, refetchResetStatus, resetStatus, siteId ] );
+
+	useEffect( () => {
+		setResetProgress( ( prevProgress ) => {
+			switch ( resetStatus ) {
+				case null:
+					return 0.1;
+				case transferStates.RELOCATING_REVERT:
+				case transferStates.ACTIVE:
+					return 0.5;
+				case transferStates.PROVISIONED:
+					return 0.6;
+				case transferStates.REVERTED:
+				case transferStates.RELOCATING:
+					return 0.85;
+				default:
+					return prevProgress + 0.05;
+			}
+		} );
+	}, [ resetStatus ] );
```

Comment:
We don't need an effect for that. With `useMemo` we can return derived values too. Maybe it's not even needed to `useMemo` - just create a function :)